File: backend/app/schemas/pagination.py
# app/schemas/pagination.py
# --- NEW FILE ---

from pydantic import BaseModel, Field
from typing import List, TypeVar, Generic

T = TypeVar('T') # Define a generic type variable

class PaginatedResponse(BaseModel, Generic[T]):
    total: int = Field(..., description="Total number of items available")
    page: int = Field(..., description="Current page number")
    limit: int = Field(..., description="Number of items per page")
    items: List[T] = Field(..., description="List of items on the current page")

File: frontend/src/types/Pagination.ts
// src/types/Pagination.ts

// This is a generic interface for paginated responses from your API.
// T represents the type of the items in the list (e.g., UserOut, InvitationOut, ClientOut).
export interface PaginatedResponse<T> {
    total: number;       // Total number of items available across all pages
    page: number;        // The current page number (usually 1-indexed)
    limit: number;       // The number of items per page
    items: T[];          // An array of items for the current page

    // Optional: If your backend provides these
    // total_pages?: number;
    // has_next?: boolean;
    // has_prev?: boolean;
}
